### PR TITLE
feat: make scaffolding cli package manager agnostic

### LINKED ISSUE #268: feat: Make scaffolding cli package manager agnostic
Currently, scaffolding ~only~ implicitly supports `npm` as the package manager when working the hApp's workspace. If the developer has a preference for a different package manager, they would need to add support for it manually, updating scripts that run scripts and add dependencies to workspace members.

This issue suggests making the cli package manager agnostic, meaning, providing the user the option to choose which package manager they would want to use while scaffolding their hApp i.e `yarn`, `pnpm` or `bun`, then ensuring all references to this script manager are used elsewhere.

### PR DESCRIPTION
**Pull request by c12i**
Closes #268 

~Depends on #313~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,37 @@
 # Holochain Scaffolding CLI
 
-CLI to easily generate and edit holochain apps.
+A command-line interface for creating and modifying a Holochain application (hApp).
```
Comment (c12i) [inline README.md:3]: @pdaoust I'd like to know whether this doc should be in the developer reference rather than here in the scaffolding cli?

**Review by matthme — CHANGES_REQUESTED** [review]
Hmm...it seems to me that the `--package-manager` option should not be at the top level but instead be an option of the `web-app` and `example` subcommands as it does not seem to have any effect beyond the initial web-app scaffolding step.
And in case that makes sense to you then I think it would be quite nice to have an interactive selection step where one can choose the package manager explicitly, akin to how the UI framework can be chosen.

**Comment by c12i** [comment]
Good idea, thought I would give it the same treatment as the `--template` flag, but you are right, the option is only relevant to `hc-scaffold web-app` and `hc-scaffold example`. Fixed.

File: README.md (hunk 1 above)
Comment (pdaoust) [reply README.md:3]: 😬 that is a very good question for which I have no answer. Ideally we'd have documentation 'wherever the dev finds themselves', so I would like to say 'yes to both places'. But that creates duplicated content. I wonder if we could write something that would inline it into the `docs-pages` build pipeline.

File: templates/custom-template/custom-template/template/web-app/package.json.hbs
Comment (pdaoust) [inline templates/custom-template/custom-template/template/web-app/package.json.hbs:1]: Does this file and the one below it in the `ui` folder also need the package manager treatment?

File: templates/headless/web-app.instructions.hbs [hunk 2]
```diff
@@ -3,16 +3,16 @@ This is a headless skeleton for a hApp.
 To build the app you can run
 
   cd {{app_name}}
-  npm install
-  npm run build:happ
+  {{(package_manager_command package_manager "install" null)}}
```
Comment (pdaoust) [inline templates/headless/web-app.instructions.hbs:6]: ~~it doesn't look like the headless one gets a `package.json` for this command to run against. Should it? it'd be convenient to have the `build:happ` command etc.~~

Wait, I'm guessing that it uses the `package.json.hbs` template from the `generic` template?

**Review by pdaoust — COMMENTED** [review]
I love all the new documentation! Just one question that may be important. Don't know whether that should be a 'comment' or 'request changes'.

File: templates/headless/web-app.instructions.hbs (hunk 2 above)
Comment (c12i) [reply templates/headless/web-app.instructions.hbs:6]: Yes, it inherits from the generic template

File: templates/custom-template/custom-template/template/web-app/package.json.hbs
Comment (c12i) [reply templates/custom-template/custom-template/template/web-app/package.json.hbs:1]: No, this is a template for creating a custom template, where the author has the freedom to pick which package manager they would wan to use

File: README.md (hunk 1 above)
Comment (c12i) [reply README.md:3]: We can publish this through Rust docs and then add a link to it in the developer guide, what do you think?

**Review by matthme — APPROVED** [review]
Great job! This PR is a nice improvement :)

**Pull request merged**